src/mlpack/methods/linear_regression/linear_regression_main.cpp

```diff
@@ -186,6 +184,10 @@ static void mlpackMain()
 
     // Save predictions.
     if (CLI::HasParam("output_predictions"))
-      CLI::GetParam<vec>("output_predictions") = std::move(predictions);
+      CLI::GetParam<rowvec>("output_predictions") = std::move(predictions);
   }
+
+  // Save the model if needed.
+  if (CLI::HasParam("output_model"))
+    CLI::GetParam<LinearRegression>("output_model") = std::move(lr);
```

Comment:
Wow, the model was also not being saved?  The fact that you found so many issues with this single program that I have hand-checked a few times really makes me wonder how many other issues will be found as we add more tests...

Reply:
The model was being saved in previous version, but it was moved to output parameters before calculating predictions so we couldn't make a prediction and creating a model at the same time

Reply:
Ah, right, sorry---I overlooked the part where it was saved previously in the diff. :)